src/main/java/com/google/api/codegen/util/java/JavaTypeTable.java

```diff
@@ -128,10 +134,15 @@ public static String getBoxedTypeName(String primitiveTypeName) {
   public Map<String, String> getImports() {
     // Clean up the imports.
     Map<String, String> cleanedImports = new TreeMap<>();
+    // Imported type is in java.lang or in package, can be ignored.
     for (String imported : imports.keySet()) {
       if (imported.startsWith(JAVA_LANG_TYPE_PREFIX)) {
-        // Imported type is in java.lang or in package, can be ignored.
         continue;
+      } else if (!implicitPackageName.isEmpty() && imported.startsWith(implicitPackageName)) {
+        // Imported type is in a subpackage must not be ignored.
+        if (imported.substring(implicitPackageName.length() + 1).split("\\.").length == 1) {
+          continue;
+        }
```

Comment:
Doing a split on a period will create new strings that won't really be used. Instead, use indexOf to see if the end of the string has a period.